File: src/1603_design_parking_system/DesignParkingSystem.java
class ParkingSystem {
    private int[] spots;

    public ParkingSystem(int big, int medium, int small) {
        spots = new int[] { 0, big, medium, small };
    }

    public boolean addCar(int carType) {
        if (spots[carType] > 0) {
            spots[carType]--;
            return true;
        }
        return false;
    }
}

public class DesignParkingSystem {
    public static void main(String[] args) {
        ParkingSystem ps = new ParkingSystem(1, 1, 0);
        System.out.println(ps.addCar(1) + " | " + true);
        System.out.println(ps.addCar(2) + " | " + true);
        System.out.println(ps.addCar(3) + " | " + false);
        System.out.println(ps.addCar(1) + " | " + false);
    }
}

File: src/1603_design_parking_system/design_parking_system.cpp
#include <iostream>
using namespace std;

class ParkingSystem {
   public:
    int spots[4];
    ParkingSystem(int big, int medium, int small) {
        spots[1] = big;
        spots[2] = medium;
        spots[3] = small;
    }
    bool addCar(int carType) {
        if (spots[carType] > 0) {
            spots[carType]--;
            return true;
        }
        return false;
    }
};

int main() {
    ParkingSystem ps(1, 1, 0);
    cout << boolalpha;
    cout << ps.addCar(1) << " | " << true << endl;
    cout << ps.addCar(2) << " | " << true << endl;
    cout << ps.addCar(3) << " | " << false << endl;
    cout << ps.addCar(1) << " | " << false << endl;
    return 0;
}
